### PR TITLE
BLD: Remove `ld_classic` usage on macOS

### LINKED ISSUE #21660: BLD, MAINT: linker warnings with newer AppleClang/ld
On latest `main` branch with `Apple clang version 16.0.0 (clang-1600.0.26.3)`, with a typical build incantation like `python dev.py build -j 16 --with-accelerate`, this warning gets spammed pretty heavily:

`ld: warning: -ld_classic is deprecated and will be removed in a future release` 

There's some discussion and a link to the official Xcode release notes section here: https://stackoverflow.com/questions/79006466/ld-classic-will-be-deprecated-in-xcode

It looks like increasing the minimum MacOS deployment target is one option, but a quick local grep suggests we have at least one usage of `export MACOSX_DEPLOYMENT_TARGET=10.9`, and IIRC messing with this stuff in our (wheel) builds has been a recipe for discovering new problems.

### PR DESCRIPTION
**Pull request by czgdp1807**
#### Reference issue

Closes https://github.com/scipy/scipy/issues/21660

#### What does this implement/fix?

It removes the usage of `ld_classic` flag on macOS. I tested locally on my system (macOS with `Apple clang version 16.0.0 (clang-1600.0.26.4)`) and on CI (see https://github.com/czgdp1807/scipy/pull/10). It works fine and doesn't give any warnings. Tests also pass.

#### Additional information

cc: @tylerjereddy @rgommers @andyfaff

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @czgdp1807. 

The concern in gh-19357 was with Clang 15, not sure if that was fixed. @adamjstewart is that still relevant for Spack you think?

**Comment by andyfaff** [comment]
The build works ok for me, but I have `clang 16.0.0` on Sequoia.
There's other warnings emitted (perhaps more visible now that ld_classic isn't spammed):
```
[198/1467] Linking target scipy/sparse/linalg/_propack/_spropack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[220/1467] Linking target scipy/sparse/linalg/_propack/_dpropack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[240/1467] Linking target scipy/sparse/linalg/_propack/_cpropack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[273/1467] Linking target scipy/sparse/linalg/_propack/_zpropack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[541/1467] Linking target scipy/sparse/linalg/_eigen/arpack/_arpack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[548/1467] Linking target scipy/stats/_mvn.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[736/1467] Linking target scipy/io/_test_fortran.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[791/1467] Linking target scipy/optimize/_cobyla.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[799/1467] Linking target scipy/optimize/_slsqp.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1272/1467] Linking target scipy/integrate/_odepack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1294/1467] Linking target scipy/integrate/_vode.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1296/1467] Linking target scipy/integrate/_dop.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1297/1467] Linking target scipy/integrate/_lsoda.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1310/1467] Linking target scipy/integrate/_test_odeint_banded.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1412/1467] Linking target scipy/interpolate/_fitpack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1429/1467] Linking target scipy/interpolate/_dfitpack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1457/1467] Linking target scipy/odr/__odrpack.cpython-313-darwin.so
ld: warning: ignoring duplicate libraries: '-lgfortran'
[1467/1467] Linking target scipy/spatial/_qhull.cpython-313-darwin.so
```

**Review by tylerjereddy — COMMENTED** [review]
Seems "ok" locally with `clang 15.0.7` from Homebrew--not sure how different that might be from the original AppleClang `15.0.0` reproducer though.. let me see if I can check..

**Comment by adamjstewart** [comment]
If you told me that the issue I reported was a bug in Spack's compiler wrappers and not in Xcode, I would not be shocked. There has been some recent discussion on the Spack side (https://github.com/spack/spack/issues/46506) that has raised concerns (or at least confusion) with my original bug report. It's possible that this was a bug in early patch releases of Xcode, or that Spack was incorrectly handling GFortran linking, or that Homebrew just has a magic patch that solves this issue.

TL;DR: I would not put much faith in my original bug report, and `ld_classic` is deprecated and will be removed someday, so I'm fine with removing this hack.

**Review by rgommers — APPROVED** [review]
Okay, sounds like this is good to go - thanks everyone!

**Comment by rgommers** [comment]
> There's other warnings emitted (perhaps more visible now that `ld_classic` isn't spammed):

I'm not seeing those on Sequoia with the conda-forge compilers, so I assume it has to do with Homebrew `gfortran`. If you want to open a new issue for it, then we can probably fix that up. The content of `build/build.ninja` for a single Fortran build target should help with that.
